Lib/fractions.py

```diff
@@ -247,12 +248,23 @@ def limit_denominator(self, max_denominator=1000000):
             n, d = d, n-a*d
 
         k = (max_denominator-q0)//q1
-        bound1 = Fraction(p0+k*p1, q0+k*q1)
-        bound2 = Fraction(p1, q1)
-        if abs(bound2 - self) <= abs(bound1-self):
-            return bound2
+        bound1_n = p0 + k*p1
+        bound1_d = q0 + k*q1
+        bound2_n = p1
+        bound2_d = q1
+        bound1_minus_self_n = abs((bound1_n * d_original)
```

Comment:
It's rare that I complain about names being too long, but I think these could be shortened. How about `diff1_n`, `diff1_d`, `diff2_n` and `diff2_d`? I'd also suggest dropping the redundant parentheses around the multiplications.